gig/views.py

```diff
@@ -114,10 +116,40 @@ def form_valid(self, form):
 
         return result
 
+class CommentsView(LoginRequiredMixin, TemplateView):
+    template_name='gig/gig_comments.html'
 
-def has_edit_permission(user, band):
-        return user.is_superuser or band.anyone_can_create_gigs or band.is_admin(user)
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        gig = Gig.objects.get(id=self.kwargs['pk'])
+        context['gig'] = gig
+        try:
+            context['comments'] = GigComment.objects.filter(gig__id=self.kwargs['pk']).order_by('created_date')
+        except GigComment.DoesNotExist:
+            context['comments'] = None
+        return context
+
+    def get(self, request, *args, **kwargs):
+        if not has_comment_permission(self.request.user, get_object_or_404(Gig, id=kwargs['pk'])):
+            raise PermissionError('Illegal access to comments')
+        return super().get(request, **kwargs)
 
+    def post(self, request, **kwargs):
+        if not has_comment_permission(self.request.user, get_object_or_404(Gig, id=kwargs['pk'])):
+            raise PermissionError('Illegal access to comments')
+
+        text = request.POST.get('commenttext','').strip()
+        if text:
+            GigComment.objects.create(text=text, member=request.user, gig=Gig.objects.get(id=kwargs['pk']))
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+
+def has_comment_permission(user, gig):
+    return Assoc.objects.filter(member = user, band=gig.band).count() == 1
```

Comment:
To make things easier, perhaps this function can trigger a 401 or 403 (again, can't remember), and then the calling functions don't need to worry about the user being unauthorized.

Reply:
agreed, bad habit